tests/functional/timeout.sh

```diff
@@ -7,39 +7,14 @@ source common.sh
 # XXX: This shouldn’t be, but #4813 cause this test to fail
 needLocalStore "see #4813"
 
-messages=$(nix-build -Q timeout.nix -A infiniteLoop --timeout 2 2>&1) && status=0 || status=$?
-
-if [ "$status" -ne 101 ]; then
-    echo "error: 'nix-store' exited with '$status'; should have exited 101"
-
-    # FIXME: https://github.com/NixOS/nix/issues/4813
-    skipTest "Do not block CI until fixed"
-
-    exit 1
-fi
-
-if echo "$messages" | grepQuietInvert "timed out"; then
-    echo "error: build may have failed for reasons other than timeout; output:"
-    echo "$messages" >&2
-    exit 1
-fi
-
-if nix-build -Q timeout.nix -A infiniteLoop --max-build-log-size 100; then
-    echo "build should have failed"
-    exit 1
-fi
-
-if nix-build timeout.nix -A silent --max-silent-time 2; then
-    echo "build should have failed"
-    exit 1
-fi
-
-if nix-build timeout.nix -A closeLog; then
-    echo "build should have failed"
-    exit 1
-fi
-
-if nix build -f timeout.nix silent --max-silent-time 2; then
-    echo "build should have failed"
-    exit 1
-fi
+# FIXME: https://github.com/NixOS/nix/issues/4813
+expectStderr 101 nix-build -Q timeout.nix -A infiniteLoop --timeout 2 | grepQuiet "timed out" \
+    || skipTest "Do not block CI until fixed"
```

Comment:
`skipTest` cancels the whole test file, which is unexpected. Easy to miss when reading this test.

`||` is for flaky tests, at best. What we should have here is a negation, so that when the behavior is fixed, the test *must* be updated to reflect the new state and actually test it.